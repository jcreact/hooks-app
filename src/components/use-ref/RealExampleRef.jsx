import React, { useState } from 'react';

import { MultipleHooks } from '../examples/MultipleHooks.jsx';

export const RealExampleRef = () => {
    const [show, setShow] = useState(false);

    return (
        <>
            <button
                className="btn btn-primary mt-3"
                onClick={() => setShow(!show)}
            >
                Mostar Citas
            </button>
            {show && <MultipleHooks />}
        </>
    );
};
