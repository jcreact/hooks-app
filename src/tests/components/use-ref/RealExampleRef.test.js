import React from 'react';
import { shallow } from 'enzyme';

import { RealExampleRef } from '../../../components/use-ref/RealExampleRef.jsx';


describe(`Pruebas en el componente 'RealExampleRef.jsx'.`, () => {

    const component = shallow(<RealExampleRef />);

    test(`Debe renderizar el componente RealExampleRef correctamente.`, () => {

        expect(component).toMatchSnapshot();
        expect(component.find('button').exists()).toBeTruthy();
        expect(component.find('button').text()).toBe('Mostar Citas');
        expect(component.find('MultipleHooks').exists()).toBeFalsy();
    });

    test(`Debe mostrar el componente 'MultipleHooks'.`, () => {
        component.find('button').simulate('click');
        expect(component.find('MultipleHooks').exists()).toBe(true);
    });

});
