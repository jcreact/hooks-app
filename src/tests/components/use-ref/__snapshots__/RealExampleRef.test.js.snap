// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente 'RealExampleRef.jsx'. Debe renderizar el componente RealExampleRef correctamente. 1`] = `
<Fragment>
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Mostar Citas
  </button>
</Fragment>
`;
